src/plugins/gds_mt/gds_mt_backend.cpp

```diff
@@ -0,0 +1,322 @@
+        if (is_local_file) {
+            base_addr = (void*)remote[i].addr;
+            if (!base_addr) {
+                delete gds_mt_handle;
+                return NIXL_ERR_INVALID_PARAM;
+            }
+            total_size = remote[i].len;
+            base_offset = (size_t)local[i].addr;
+
+            auto it = gds_mt_file_map.find(local[i].devId);
+            if (it == gds_mt_file_map.end()) {
+                NIXL_ERROR << "GDS_MT: error: file handle not found";
+                delete gds_mt_handle;
+                return NIXL_ERR_NOT_FOUND;
+            }
+            fh = it->second;
+        } else {
+            base_addr = (void*)local[i].addr;
+            if (!base_addr) {
+                delete gds_mt_handle;
+                return NIXL_ERR_INVALID_PARAM;
+            }
+            total_size = local[i].len;
+            base_offset = (size_t)remote[i].addr;
+
+            auto it = gds_mt_file_map.find(remote[i].devId);
+            if (it == gds_mt_file_map.end()) {
+                NIXL_ERROR << "GDS_MT: error: file handle not found";
+                delete gds_mt_handle;
+                return NIXL_ERR_NOT_FOUND;
+            }
+            fh = it->second;
+        }
+
+        GdsMtTransferRequestH req;
+        req.addr = (char*)base_addr;
+        req.size = total_size;
+        req.file_offset = base_offset;
+        req.fh = fh.cu_fhandle;
+        req.op = (operation == NIXL_READ) ? CUFILE_READ : CUFILE_WRITE;
+        gds_mt_handle->request_list.push_back(req);
+    }
+
+    if (gds_mt_handle->request_list.empty()) {
+        delete gds_mt_handle;
+        return NIXL_ERR_INVALID_PARAM;
+    }
+    for (GdsMtTransferRequestH& req : gds_mt_handle->request_list) {
+        GdsMtTransferRequestH* captured_req = &req;
+        gds_mt_handle->taskflow.emplace([gds_mt_handle, captured_req]() {
+            worker_thread(gds_mt_handle, captured_req);
+        });
+    }
+
+
+    gds_mt_handle->needs_prep = false;
+    handle = gds_mt_handle;
+    return NIXL_SUCCESS;
+}
+
+nixl_status_t nixlGdsMtEngine::postXfer(const nixl_xfer_op_t &operation,
+                                        const nixl_meta_dlist_t &local,
+                                        const nixl_meta_dlist_t &remote,
+                                        const std::string &remote_agent,
+                                        nixlBackendReqH* &handle,
+                                        const nixl_opt_b_args_t* opt_args) const
+{
+    nixlGdsMtBackendReqH* gds_mt_handle = (nixlGdsMtBackendReqH*)handle;
+    if (!gds_mt_handle) {
+        NIXL_ERROR << "GDS_MT: error: invalid handle";
+        return NIXL_ERR_INVALID_PARAM;
+    }
+    if (gds_mt_handle->request_list.empty()) {
+        NIXL_ERROR << "GDS_MT: error: empty request list for Xfer";
+        return NIXL_ERR_INVALID_PARAM;
+    }
+    gds_mt_handle->running_transfer = executor->run(gds_mt_handle->taskflow);
```

Comment:
@sdimitro : Will the executor->run work properly when the same BackendReqH  is reused during reposts?